frontend/src/utilities/utils.ts

```diff
@@ -68,7 +68,7 @@ export const getLabelColorForDocType = (
       return 'grey';
   }
 };
-export const combineCategoryAnnotations = (doc: OdhDocument, app: OdhApplication): void => {
+export const combineCategoryAnnotations = (doc: OdhDocument, app: OdhApplication): OdhDocument => {
```

Comment:
You cannot simply change the behavior of a function like this without addressing the caller.
I think there's a bug in the code calling this function.

Best to add a new value to the rule option support ignoring the rule.
`"ignorePropertyModificationsForRegex": ["^assignable[A-Z]"]`

Then rename `doc` to `assignableDoc`.